pkg/cmd/server/start.go

```diff
@@ -564,6 +592,21 @@ func defaultMasterAddress(cfg *config) error {
 	return nil
 }
 
+// clientConfigFromKubeConfig reads the client configuration settings for connecting to
+// a Kubernetes master.
+func clientConfigFromKubeConfig(cfg *config) *kclient.Config {
+	config, err := cfg.ClientConfig.ClientConfig()
+	if err != nil {
+		glog.Fatalf("Unable to read client configuration: %v", err)
+	}
+	if len(config.Version) == 0 {
+		config.Version = klatest.Version
+	}
+	kclient.SetKubernetesDefaults(config)
+	config.Host = cfg.KubernetesAddr.URL.String()
```

Comment:
Yeah - but cli Args always override files.  I guess we might check it for kubernetes.


> On Jan 29, 2015, at 8:17 PM, Jordan Liggitt <notifications@github.com> wrote:
> 
> In pkg/cmd/server/start.go:
> 
> > @@ -564,6 +592,21 @@ func defaultMasterAddress(cfg *config) error {
> >  	return nil
> >  }
> >  
> > +// clientConfigFromKubeConfig reads the client configuration settings for connecting to
> > +// a Kubernetes master.
> > +func clientConfigFromKubeConfig(cfg *config) *kclient.Config {
> > +	config, err := cfg.ClientConfig.ClientConfig()
> > +	if err != nil {
> > +		glog.Fatalf("Unable to read client configuration: %v", err)
> > +	}
> > +	if len(config.Version) == 0 {
> > +		config.Version = klatest.Version
> > +	}
> > +	kclient.SetKubernetesDefaults(config)
> > +	config.Host = cfg.KubernetesAddr.URL.String()
> Reading most things from .kubeconfig, then trumping with the command-line host seems hard to reason about
> 
> —
> Reply to this email directly or view it on GitHub.
>